### DISCUSSION TITLE
str.to_code function

### DISCUSSION THREAD

**Discussion by Avgor46** [sosy-lab/java-smt#420] [Q&A] [answered]
Hi!

It seems that CVC5 supports `str.to_code` function, but I cannot find such method in any `StringFormulaManager` (CVC5 and other). Why is this function not being processed at the moment?

**Comment by kfriedberger** [comment]
It looks like nobody has yet requested support for `string::to_code` from JavaSMT. We could add this in the next release, if required.

**Comment by kfriedberger** [comment]
Links to solver documentation: 
- Z3: https://z3prover.github.io/api/html/group__capi.html#gaf7f1141487f7fb41beca2d469e9d633f
- CVC5: https://cvc5.github.io/docs/cvc5-1.1.1/api/cpp/kind.html#_CPPv4N4cvc54Kind14STRING_TO_CODEE

With `string::to_code` we should also add `string::from_code`.

**Comment by Avgor46** [comment]
That would be very helpful!

**Comment by kfriedberger** [comment]
Work in progress: https://github.com/sosy-lab/java-smt/pull/422

**Comment by kfriedberger** [comment, marked as answer]
https://github.com/sosy-lab/java-smt/pull/422 was merged and will be part of the next release.
